stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -456,8 +456,7 @@ function ldiv!(D::Diagonal, A::Union{LowerTriangular,UpperTriangular})
     A
 end
 
-function rdiv!(A::AbstractMatrix{T}, D::Diagonal{T}) where {T}
-    require_one_based_indexing(A)
+function rdiv!(A::AbstractMatrix, D::Diagonal)
```

Comment:
We need to require one-based indexing, because we are indexing into `A` with one-based ranges.
```suggestion
function rdiv!(A::AbstractMatrix, D::Diagonal)
    require_one_based_indexing(A)
```